rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java

```diff
@@ -526,28 +535,59 @@ public Optional<Block> getLatestBlock() {
         return Optional.ofNullable(latestBlock);
     }
 
-    class NewBlockListener extends EthereumListenerAdapter {
-
-        @Override
+    static class NewBlockTxListener extends EthereumListenerAdapter {
```

Comment:
This is not private just for it to be accessible for testing, right? This is not ideal, but if you are not able to avoid it please add at least a `@VisibleForTesting` so Sonar will complain if we use it anywhere else.

Reply:
done